scala3doc/src/main/scala/dotty/dokka/tasty/TastyParser.scala

```diff
@@ -0,0 +1,205 @@
+
+  end TastyFromClass
+
+  class TastyInspectorPhase extends Phase:
+
+    override def phaseName: String = "tastyInspector"
+
+    override def run(implicit ctx: Context): Unit =
+      val qctx = QuoteContextImpl()
+      self.processCompilationUnit(using qctx)(ctx.compilationUnit.tpdTree.asInstanceOf[qctx.tasty.Tree])
+
+  end TastyInspectorPhase
+
+end SbtDokkaTastyInspector
+
+trait DokkaBaseTastyInspector:
+  val sourceSet: SourceSetWrapper
+  val parser: Parser
+  val config: DottyDokkaConfig
+
+  private val topLevels = Seq.newBuilder[Documentable]
+
+  def processCompilationUnit(using ctx: quoted.QuoteContext)(root: ctx.reflect.Tree): Unit =
+    val parser = new TastyParser(ctx.reflect, this, config)
+    topLevels ++= parser.parseRootTree(root.asInstanceOf[parser.reflect.Tree])
+
+  def result(): List[DPackage] =
+    val all = topLevels.result()
+    val packages = all
+      .filter(_.isInstanceOf[DPackage])
+      .map(_.asInstanceOf[DPackage])
+      .groupBy(_.getDri)
+      .map((dri, pckgs) =>
+        pckgs.reduce(_.mergeWith(_))
+      )
+
+    val byPackage = all.filter(_.getDri != null).groupBy(_.getDri().getPackageName())
+    byPackage.map {
+      case (pck, entries) => {
+        val found = packages.find(d => d.getName == pck)
+        .map( f =>
+          new DPackage(
+            f.getDri,
+            f.getFunctions,
+            f.getProperties,
+            Nil.asJava, // TODO add support for other things like type or package object entries
+            Nil.asJava,
+            f.getDocumentation,
+            null,
+            sourceSet.toSet,
+            f.getExtra
+          ).withNewMembers(entries.filterNot(_.isInstanceOf[DPackage]).toList).asInstanceOf[DPackage]
+        )
+        found.getOrElse(throw IllegalStateException("No package for entries found"))
+      }
+    }.toList
+
+  extension (self: DPackage) def mergeWith(other: DPackage): DPackage =
+    val doc1 = self.getDocumentation.asScala.get(sourceSet.getSourceSet).map(_.getChildren).getOrElse(Nil.asJava)
+    val doc2 = other.getDocumentation.asScala.get(sourceSet.getSourceSet).map(_.getChildren).getOrElse(Nil.asJava)
+    mergeExtras(
+      DPackage(
+        self.getDri,
+        (self.getFunctions.asScala ++ other.getFunctions.asScala).asJava,
+        (self.getProperties.asScala ++ other.getProperties.asScala).asJava,
+        Nil.asJava, // WARNING Merging is done before collecting classlikes, if it changes it needs to be refactored
+        Nil.asJava,
+        sourceSet.asMap(
+            DocumentationNode(
+                (
+                  doc1.asScala ++ doc2.asScala
+                ).asJava
+            )
+        ),
+        null,
+        sourceSet.toSet,
+        PropertyContainer.Companion.empty()
+      ),
+      self, 
+      other
+    )
+
+/** Parses a single Tasty compilation unit. */
+case class TastyParser(reflect: Reflection, inspector: DokkaBaseTastyInspector, config: DottyDokkaConfig)
```

Comment:
```suggestion
case class TastyParser(qctx: QuoteContext, inspector: DokkaBaseTastyInspector, config: DottyDokkaConfig)
```

We should always use the full path for reflection. Using `Reflection` directly may not work in the future.

Reply:
Adressed in 78b7d4976324d65bea063e4b3718c61f140aac04 in #10092